tests/test_warp.py

```diff
@@ -1286,3 +1286,24 @@ def test_reproject_dst_alpha(path_rgb_msk_byte_tif):
             dst_alpha=4)
 
         assert dst_arr[3].any()
+
+
+def test_issue1350():
+    """Warp bands other than 1 or All"""
+
+    with rasterio.open('tests/data/RGB.byte.tif') as src:
+        dst_crs = {'init': 'EPSG:3857'}
+        out1 = np.zeros(src.shape, dtype=src.dtypes[0])
+        out2 = np.zeros(src.shape, dtype=src.dtypes[1])
+        out3 = np.zeros(src.shape, dtype=src.dtypes[2])
+
+        for i, out in enumerate([out1, out2, out3]):
+            reproject(
+                rasterio.band(src, i+1),
+                out,
+                resampling=Resampling.nearest,
+                dst_transform=DST_TRANSFORM,
+                dst_crs=dst_crs)
+
+        assert not (out1 == out2).all()
+        assert not (out2 == out3).all()
```

Comment:
@Kirill888 this test is failing with GDAL 2.1 and 2.2. I suggest the following test:

```python
def test_issue1350():
    """Warp bands other than 1 or All"""

    with rasterio.open('tests/data/RGB.byte.tif') as src:
        dst_crs = {'init': 'EPSG:3857'}

        reprojected = []

        for dtype, idx in zip(src.dtypes, src.indexes):

            out = np.zeros(src.shape, dtype=dtype)

            reproject(
                rasterio.band(src, idx),
                out,
                resampling=Resampling.nearest,
                dst_transform=DST_TRANSFORM,
                dst_crs=dst_crs)

            reprojected.append(out)

        assert not (reprojected[0] == reprojected[1]).all()
        assert not (reprojected[1] == reprojected[2]).all()
```

Note that in _warp.pyx, when `numpy.reshape()` is called, we get a new object that is no longer referenced by any element of `[out1, out2, out3]`. Thus we must aggregate it after calling `reproject()`.

I'm probably going to change `reproject()` to return the destination in the same way that numpy functions do, but the test change above will be fine for now.

Reply:
Thanks @sgillies I made the change and rebased against `maint-1.0`

So does this mean that to be certain that `reproject` writes to supplied memory you have provide `ndarray` with `ndim==3`?

Reply:
Yes, that's right.

Reply:
Unfortunately the problem is somewhere else, those GDAL versions are still failing after the change to the test code.